src/client/Client.cc

```diff
@@ -6221,9 +6222,65 @@ void Client::tick()
     check_caps(in, CHECK_CAPS_NODELAY);
   }
 
+  collect_and_send_metrics();
+
   trim_cache(true);
 }
 
+void Client::collect_and_send_metrics() {
+  ldout(cct, 20) << __func__ << dendl;
+
+  ceph_assert(ceph_mutex_is_locked_by_me(client_lock));
+
+  bool send_global_metrics = true;
+  for (auto &p : mds_sessions) {
+    ClientMetricMessage metric;
+    std::vector<ClientMetricMessage> message;
+
+    ldout(cct, 20) << __func__ << ": rank=" << p.first << dendl;
+
+    if (send_global_metrics) {
+      send_global_metrics = false;
+      collect_global_metrics(&message);
+    }
+
+    metric = collect_cap_metric(p.second);
+    message.push_back(metric);
+
+    p.second.con->send_message2(make_message<MClientMetrics>(std::move(message)));
+  }
+}
+
+void Client::collect_global_metrics(std::vector<ClientMetricMessage> *message) {
+  ldout(cct, 20) << __func__ << dendl;
+
+  ceph_assert(ceph_mutex_is_locked_by_me(client_lock));
+
+  ClientMetricMessage metric;
+  // read latency
+  metric = ClientMetricMessage(ReadLatencyPayload(logger->tget(l_c_read)));
```

Comment:
@vshankar 

BTW， this will collect the sum time instead of the avg time.
Should it be better to collect and send the avg time ?

Reply:
I reused the counter that's already tracked. We can start tracking the averages and start sending those.

Reply:
I assumed we should send the avg ones, so currently in kclient fulfillment I am sending the avg latencies.

Reply:
@batrick @lxbsz  -- we should finalize the default set of metrics since pr https://github.com/ceph/ceph/pull/26004 is getting close.

right now we have cap hit metric (per mds) and some global metrics (read/write/metadata latencies). we can stick to these for now and get these counters accurate.

Reply:
The average and sum of read latency are both useful. Let's standardize on the sum here. We can refactor the MDS struct names later to clarify.

Reply:
And also should we record the min/max latency ?

Reply:
> And also should we record the min/max latency ?

MDS should be able to figure that out.

Reply:
We can figure the metadata's min/max latency out in MDS, but they will not consider the network latency between MDS and client, like the sum latency we do for metadata.

And for the read/write requests they will bypass the MDS, do you mean the MDS will collect them from OSD ?

Reply:
I meant min/max from the set of average latencies sent by the client. I think you mean min/max latency rather than the average -- in that case, yes, the client would need to additionally send that.

Reply:
Yeah, right. We were not on the same page :-).